src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,222 @@
+            {
+                var failureMessage = (EpisodeImportFailedEvent)message;
+
+                // if we can see the file exists but the import failed then likely a permissions issue
+                if (failureMessage.EpisodeInfo != null)
+                {
+                    var episodePath = failureMessage.EpisodeInfo.Path;
+                    if (_diskProvider.FileExists(episodePath))
+                    {
+                        return new HealthCheck(GetType(), HealthCheckResult.Error, string.Format("Sonarr can see but not access downloaded episode {0}.Likely permissions error.", episodePath), "#permissions-error");
+                    }
+                    else
+                    {
+                        // If the file doesn't exist but EpisodeInfo is not null then the message is coming from
+                        // ImportApprovedEpisodes and the file must have been removed part way through processing
+                        return new HealthCheck(GetType(), HealthCheckResult.Error, string.Format("File {0} was removed part way through processing.", episodePath), "#remote-path-file-removed");
+                    }
+                }
+
+                // If the previous case did not match then the failure occured in DownloadedEpisodeImportService,
+                // while trying to locate the files reported by the download client
+                // Only check clients not in failure status, those get another message
+                var client = _downloadClientProvider.GetDownloadClients().FirstOrDefault(x => x.Definition.Name == failureMessage.DownloadClientInfo.Name);
+
+                if (client == null)
+                {
+                    return new HealthCheck(GetType());
+                }
+
+                try
+                {
+                    var status = client.GetStatus();
+                    var dlpath = client?.GetItems().FirstOrDefault(x => x.DownloadId == failureMessage.DownloadId)?.OutputPath.FullPath;
+
+                    // If dlpath is null then there's not much useful we can report.  Give a generic message so
+                    // that the user realises something is wrong.
+                    if (dlpath.IsNullOrWhiteSpace())
+                    {
+                        return new HealthCheck(GetType(), HealthCheckResult.Error, "Sonarr failed to import (an) episode(s).  Check your logs for details.", "#remote-path-import-failed");
+                    }
+
+                    if (!dlpath.IsPathValid())
+                    {
+                        if (!status.IsLocalhost)
+                        {
+                            return new HealthCheck(GetType(), HealthCheckResult.Error, string.Format("Remote download client {0} reported files in {1} but this is not a valid {2} path.  Review your remote path mappings and download client settings.", client.Definition.Name, dlpath, _osInfo.Name), "#bad-remote-path-mapping");
```

Comment:
What's with all the double spaces between sentences? Please update to a single space